### PR TITLE
Fix Violations of and Reenable `Rails/Date`

### PR DESCRIPTION
**Pull request by Hamms**
> This cop checks for the correct use of Date methods, such as Date.today, Date.current etc.
>
> Using Date.today is dangerous, because it doesn't know anything about Rails time zone. You must use Time.zone.today instead.

Fix applied mostly with sed in vim: `%s/Date\.\(today\|current\|yesterday\|tomorrow\)/Time.zone.\1/gc`

I will note that it is theoretically possible that we are for some reason relying on time-zone-agnostic functionality for some subset of the affected code. I manually inspected all changes for any obvious instances of this strange assumption and didn't find any, but this is a large changeset and I easily could have missed something. I think this is still worth doing for consistency's sake, but this should be considered a medium-risk change.

## Testing Story

Relying on existing tests to verify that this doesn't represent any change in functionality. Open to feedback on any better testing I could do!

## Links

- https://www.rubydoc.info/gems/rubocop/0.41.2/RuboCop/Cop/Rails/Date
- https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsdate

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by bethanyaconnor — COMMENTED** [review]
The change looks good to me but I'd like @code-dot-org/acquisition-products-team to take a look as the vast majority of these changes are in their space

**Comment by megcrenshaw** [comment]
@TurnerRiley lots of these changes are in regards to workshops, and you know the specs there better than I do. Could you take a look at this? I believe this will ensure we are looking at all times in UTC, rather than trying to be time zone agnostic. That sounds reasonable to me, but would love your eyes on it.

File: dashboard/test/models/regional_partner_test.rb [hunk 1]
```diff
@@ -190,13 +190,13 @@ class RegionalPartnerTest < ActiveSupport::TestCase
     create :regional_partner_program_manager, regional_partner: regional_partner, program_manager: partner_organizer
 
     future_partner_workshops = [
-      create(:workshop, organizer: partner_organizer, sessions_from: Date.today),
+      create(:workshop, organizer: partner_organizer, sessions_from: Time.zone.today),
       create(:workshop, organizer: partner_organizer, sessions_from: Date.tomorrow)
```
Comment (TurnerRiley) [inline dashboard/test/models/regional_partner_test.rb:194]: Why was "Date" in the first line changed but not the second? If there's no `Time.zone.tomorrow` function could we do `Time.zone.today + 1.day`?

**Review by TurnerRiley — APPROVED** [review]
This looks good to me! Newer tests in the workshop area have used `Time.now` so I like this idea of unifying it all! One nit that I think could be indicative of other uses of "Date" potentially? I imagine the difference is not too severe as long as tests are still passing.

File: dashboard/test/models/regional_partner_test.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/test/models/regional_partner_test.rb:194]: Oh, good question, thanks for asking; I forgot to update the PR description for this. The default `EnforcedStyle` option for this is `flexible`, which means only instances of `Date.today` are prohibited even though `Time.zone.tomorrow` and others do indeed exist. Originally I did indeed replace all instances of `current`, `yesterday`, and `tomorrow` as well, but ran into some issue and decided to scope this change to just what the rule requires by default.

I think I'd be in favor of us setting `EnforcedStyle` to  `strict` and updating everything else, but decided to save that for follow-up work

**Pull request merged**